### DISCUSSION TITLE
Feature Request: Option to use field names in permalinks

### DISCUSSION THREAD

**Discussion by fireundubh** [pods-framework/pods#6668] [Ideas]
I'd like to be able to use the unique name of a field belonging to a post type or taxonomy in a permalink.

This is one way this would be done with filters and a relationship field:

``` php
if (!function_exists('employer_permalink'))
{
    add_filter('post_link', 'employer_permalink', 10, 3);
    add_filter('post_type_link', 'employer_permalink', 10, 3);
    function employer_permalink($permalink, $post, $leavename)
    {
        if (!$post) return $permalink;

        if (strpos($permalink, '%job_employer%') === FALSE) return $permalink;

        // in this example, "job_employer" is a required relationship field
        $related_post = pods('job', $post->ID)->field('job_employer');

        return str_replace('%job_employer%', $related_post['post_name'], $permalink);
    }
}

// you have to update the WP Permalinks page to force this change to take effect
if (!function_exists('employer_rewrite_tag'))
{
    add_action('init', 'employer_rewrite_tag', 10, 0);
    function employer_rewrite_tag() {
        add_rewrite_tag('%job_employer%', '([^&]+)');
    }
}
```

You could add a checkbox to the Options or Advanced sections of the Edit Field form. When checked, the user could use the unique field name as a tag in permalinks. The user could also use a plugin like [Custom Post Type Permalinks](https://wordpress.org/plugins/custom-post-type-permalinks/) to fully customize the permalinks for Pods post types.

**Comment by JoryHogeveen** [comment]
Though @sc0ttkclark will have to look at this.. I think this is currently out of scope for 2.x.
It will require a lot of filters and testing to get something like this stable. 
Though it would be interesting to look into this (I've had a similar situation once).

I'll bundle some doc's here:
https://codex.wordpress.org/Class_Reference/WP_Rewrite (main rewrite class)
https://codex.wordpress.org/Plugin_API/Filter_Reference/post_link (generate permalink filter)
https://codex.wordpress.org/Plugin_API/Filter_Reference/post_type_link (generate permalink filter)
https://codex.wordpress.org/Function_Reference/add_permastruct
https://codex.wordpress.org/Rewrite_API/add_rewrite_tag
https://codex.wordpress.org/Rewrite_API/add_rewrite_rule
https://codex.wordpress.org/Rewrite_API/flush_rules (after pod save)

**Comment by MRamonLeon** [comment]
me too
